src/errors/InvalidNameError.ts

```diff
@@ -0,0 +1,12 @@
+import { Annotated } from './Annotated';
+
+export class InvalidNameError extends Error implements Annotated {
+  specReferences = [
+    'http://hl7.org/fhir/R4/structuredefinition-definitions.html#StructureDefinition.name',
+    'http://hl7.org/fhir/R4/valueset-definitions.html#ValueSet.name',
+    'http://hl7.org/fhir/R4/codesystem-definitions.html#CodeSystem.name'
+  ];
+  constructor(public badName: string) {
+    super(`The string "${badName}" does not represent a valid FHIR name.`);
```

Comment:
Similar to above, I think it would be helpful to provide the user more guidance.  FHIR doesn't spell this one out in narrative as well as ID, but following the form of how they define id, maybe we could say something like:

> The string "${badName}" does not represent a valid FHIR name.  This name should be usable as an identifier for the module by machine processing applications such as code generation.  Valid names start with an upper-case ASCII letter ('A'..'Z') followed by any combination of of upper- or lower-case ASCII letters ('A'..'Z', and 'a'..'z'), numerals ('0'..'9') and '_', with a length limit of 255 characters.

Reply:
I agree that providing additional information is the right decision here. The full explanation of a valid id or name is pretty long, so do you think it would suffice to provide a link to the relevant page in the FHIR documentation? On the other hand, it's nice to not have to look somewhere else for the answer and just have SUSHI provide the user with all the information they need. I'll consider the options over the weekend.

Reply:
I think we can remove "This name should be usable as an identifier for the module by machine processing applications such as code generation" from above, but I think it's otherwise helpful to have the info right there, even if it is a little long.  That's my opinion anyway -- but I don't feel strongly enough about it that I'd object if you thought otherwise.